### PR TITLE
Build an unsigned rpm in dev pipeline

### PR DESCRIPTION
**Pull request by mellery451**

### CODE REVIEW

**Comment by mellery451** [comment]
the RPM build failure is expected in these results until we merge https://github.com/ripple/rippled/pull/2475

File: Jenkinsfile [hunk 1]
```diff
@@ -342,6 +249,50 @@ ccache -s
...
 
+        // Also add a single build job for doing the RPM build
+        // on a docker node
+        builds['rpm'] = {
+            node('docker') {
+                def bldlabel = 'rpm'
+                configFileProvider (
+                    [configFile(
+                        fileId: 'rippled-commit-signer-public-keys.txt',
+                        variable: 'SIGNER_PUBLIC_KEYS')])
```
Comment (wilsonianb) [inline Jenkinsfile]: Thinking out loud:
We can exclude this file and allow an rpm to be built for commits from external developers
https://github.com/ripple/rippled-package-builder/blob/5578d26221e294109fdf79e0d75f90ed0bdb9f2e/rpm-builder/build_rpm.sh#L34
However, I think I prefer using the file so we don't get surprised trying to build an rpm after merging a commit missing a signature from a whitelisted key.

File: Jenkinsfile [hunk 2]
```diff
@@ -536,4 +510,192 @@ def upDir(path) {
...
+
+cat > test_rpm.sh << "EOL"
+#!/bin/bash
+
+function error {
+  echo $1
+  exit 1
+}
+
+yum groupinstall -y "Development Tools"
```
Comment (wilsonianb) [inline Jenkinsfile]: I don't think we need `"Development Tools"`

File: Jenkinsfile [hunk 3]
```diff
@@ -536,4 +510,192 @@ def upDir(path) {
...
+yum install -y yum-utils
+rpm -i /opt/rippled-rpm/*.rpm
+rc=$?; if [[ $rc != 0 ]]; then
+  error "error installing rpms"
+fi
+
+rpm -i rippled-*.src.rpm
+tar -zxf ~/rpmbuild/SOURCES/rippled.tar.gz -C ./
```
Comment (wilsonianb) [inline Jenkinsfile]: The src rpm was deleted earlier, so these two lines should be removed.

File: Jenkinsfile [hunk 4]
```diff
@@ -536,4 +510,192 @@ def upDir(path) {
...
+  error "validator-keys --unittest failed"
+fi
+
+EOL
+
+chmod +x test_rpm.sh
+
+echo "Running test container"
+
+sudo docker run --rm \
```
Comment (wilsonianb) [inline Jenkinsfile]: We run `docker` without `sudo` above. Should we do the same here?

File: Jenkinsfile [hunk 5]
```diff
@@ -342,6 +249,50 @@ ccache -s
...
+                                sh "if [ -d rpm-out ]; then rm -rf rpm-out; fi"
+                                sh rpmBuildCmd(bldlabel)
+                            }
+                            finally {
+                                def st = reportStatus(bldlabel, bldlabel, env.BUILD_URL)
+                                lock('rippled_dev_status') {
+                                    all_status[bldlabel] = st
+                                }
+                                archiveArtifacts(
+                                    artifacts: 'rpm-out/*.rpm',
```
Comment (wilsonianb) [inline Jenkinsfile]: Should we only archive the binary rpm? The src rpm can be >100 MB.

**Comment by mellery451** [comment]
@wilsonianb ..test in what way? The pipeline is running the build on this branch, the logfile is listed in results (https://ci.ops.ripple.com/job/rippled-dev/job/PR-2494/1//artifact/rpm.txt). What else did you have in mind?

File: Jenkinsfile [hunk 6]
```diff
@@ -536,4 +510,193 @@ def upDir(path) {
...
+  exit 1
+}
+
+yum install -y yum-utils
+rpm -i /opt/rippled-rpm/*.rpm
+rc=$?; if [[ $rc != 0 ]]; then
+  error "error installing rpms"
+fi
+
+/opt/ripple/bin/rippled --unittest
```
Comment (wilsonianb) [inline Jenkinsfile]: You could temporarily change this to something like `--unittest=ValidatorKeys` or rebase on Scott's fix to verify that we make it through the rpm build when tests pass.

**Pull request closed without merging**